### ISSUE TITLE
iOS ScaleBar is not movable via contentInset

### ISSUE THREAD

**Issue by racer1988** [mapbox/mapbox-gl-native#14487]
Hello.

This is similar to #11068 and would like to extend #13911
```
[self.scaleBarConstraints addObject:[self constraintForYAxisAnchor:self.scaleBar.topAnchor belowAnchor:safeAreaLayoutGuide.topAnchor]];
```

The scale bar is attached to the topAnchor.
It is not possible to customize the position of the scalebar with the `contentInset` or any other property.

The scalebar does react to left content inset tough.
```
        [self.scaleBarConstraints addObject:[self.scaleBar.leftAnchor constraintEqualToAnchor:safeAreaLayoutGuide.leftAnchor
                                                                                     constant:8.0 + self.contentInset.left]];
```

How could a scaleBar currently be moved from the top edge?

I would like to propose to change the top constraint of all views (compass, scalebar etc) to follow the inset

```
[self.scaleBarConstraints addObject:[self constraintForYAxisAnchor:self.scaleBar.topAnchor belowAnchor:safeAreaLayoutGuide.topAnchor constant:self.contentInset.top]];
```

or, if that has unexpected result, to create a new property to inset the accessories

**Comment by racer1988** [comment]
question: why this was removed in #10658

**Comment by racer1988** [comment]
To my understanding this is fixed in ios-v4.10.0 by  (#13911)

**Comment by julianrex** [comment]
> To my understanding this is fixed in ios-v4.10.0 by (#13911)

@racer1988 that's correct. Please let us know if you have an issues with this approach.

**Comment by racer1988** [comment]
@julianrex I found a issue with the default margin, you can find it here: #14586
